pkg/devicetemplateconverter/version_profile_fetcher.go

```diff
@@ -0,0 +1,74 @@
+// Copyright © 2022 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package devicetemplateconverter
+
+import (
+	"context"
+
+	"go.thethings.network/lorawan-stack/v3/pkg/cluster"
+	"go.thethings.network/lorawan-stack/v3/pkg/log"
+	"go.thethings.network/lorawan-stack/v3/pkg/rpcmetadata"
+	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+	"google.golang.org/grpc"
+)
+
+// Component abstracts the underlying *component.Component.
+type Component interface {
+	GetPeerConn(ctx context.Context, role ttnpb.ClusterRole, ids cluster.EntityIdentifiers) (*grpc.ClientConn, error)
+	AllowInsecureForCredentials() bool
+}
+
+// VersionIDProfileFetcher handles the validation and fetching of end device profile in the Device Repository.
+type VersionIDProfileFetcher struct {
+	Component        Component
+	DeviceRepository ttnpb.DeviceRepositoryClient
+}
+
+// ShouldFetchProfile dictactes if the end device has the necessary fields to fetch its profile.
+func (*VersionIDProfileFetcher) ShouldFetchProfile(device *ttnpb.EndDevice) bool {
+	return device.GetVersionIds().GetBrandId() != "" &&
+		device.GetVersionIds().GetModelId() != "" &&
+		device.GetVersionIds().GetHardwareVersion() != "" &&
```

Comment:
The hardware version is actually not required by the device repository. For example Dragino LHT65 does not have one. 
As such, you may skip it from the check here I believe.